solr/solrj/src/java/org/apache/solr/common/util/PathTrie.java

```diff
@@ -82,6 +80,22 @@ public boolean add(String s) {
     return parts;
   }
 
+  public T unregister(List<String> path) {
```

Comment:
I think the name "unregister" would only make sense if there was a corresponding "register" method.  How about a more generic "remove"?

Reply:
yeah, `remove()` is better for `insert()`